build_tools/python/e2e_test_framework/definitions/common_definitions.py

```diff
@@ -122,6 +131,29 @@ class DeviceSpec(object):
   # these parameters to set up the devices. E.g. set CPU mask.
   device_parameters: List[str] = dataclasses.field(default_factory=list)
 
+  def __str__(self):
+    return self.name
+
+  @staticmethod
```

Comment:
Nit: I would call these `@classmethod`. I think that makes more sense for factory methods. You can use `cls` instead of explicitly calling the name (not that it makes a difference here).

Reply:
Done.